### ISSUE TITLE
[TW-1401] propagating wait property to sub-tasks and/or filter for excluding sub-tasks of waiting tasks

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1428]
_Stefan Frühwirth on 2014-08-26T15:45:33Z says:_ 

 For some use cases it would be nice to "transfer" the wait property to sub-tasks (i.e. task depending on the "waiting" task) so that these tasks are hidden/excludable too (e.g. in next reports).

Taskwarrior should recognize that there is a task that can't be done "at the moment" (i.e. the point in time one requests a report) because of an active wait date, and therefore knows that tasks that are blocked by the waiting task can't be done either and that there is no point in including it in the next report.

Use case: I have to wait for team members to review or test work I've done. Some things, like releasing a feature on production servers, can't be completed until someone tested it and gave feedback. I then could create create a (pseudo)task "testing xyz (team)" and let it wait a week or so. When its done I would automatically unblock the dependent tasks.

Additionally, and ideally, there would be a way to mark a task as "waiting for xyz to happen" (as opposed to an expiration date), having this displayed in (next) reports and/or having the task's priority decreased.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-08-26T15:45:33Z
Modified: 2015-09-02T00:17:30Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-12-28T02:00:23Z says:_ 

 This issue is two separate concerns.

First, propagating more attributes to recurring task instances, but with relative dates, is planned and generally referred to as 'deferred eval'.

Second, waited tasks are given a lower urgency, with a configurable coefficient.  This moves the task lower on the 'next' report, which is sorted by urgency.  If you have a waiting task on your 'next' report, then the coefficient is not high enough, and/or there aren't enough higher-urgency tasks.  This is exactly how urgency is intended to work - the interplay of competing urgencies, high and low, positive and negative.  Explicitly keeping a task off the 'next' report is not something we want to do.
